Model/Menu/ImportProcessor/Menu.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Snowdog\Menu\Model\Menu\ImportProcessor;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\ValidatorException;
+use Snowdog\Menu\Api\Data\MenuInterface;
+use Snowdog\Menu\Api\Data\MenuInterfaceFactory;
+use Snowdog\Menu\Api\MenuRepositoryInterface;
+use Snowdog\Menu\Model\Menu\ExportProcessor;
+
+class Menu
+{
+    const REQUIRED_FIELDS = [
+        MenuInterface::TITLE,
+        MenuInterface::IDENTIFIER,
+        MenuInterface::CSS_CLASS,
+        MenuInterface::IS_ACTIVE,
+        ExportProcessor::STORES_CSV_FIELD
+    ];
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var MenuInterfaceFactory
+     */
+    private $menuFactory;
+
+    /**
+     * @var MenuRepositoryInterface
+     */
+    private $menuRepository;
+
+    public function __construct(
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        MenuInterfaceFactory $menuFactory,
+        MenuRepositoryInterface $menuRepository
+    ) {
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->menuFactory = $menuFactory;
+        $this->menuRepository = $menuRepository;
+    }
+
+    /**
+     * @return MenuInterface
+     */
+    public function createMenu(array $data, array $stores)
+    {
+        $menu = $this->menuFactory->create();
+        $data[MenuInterface::IDENTIFIER] = $this->getNewMenuIdentifier($data[MenuInterface::IDENTIFIER]);
+
+        $menu->setData($data);
+        $this->menuRepository->save($menu);
+        $menu->saveStores($stores);
+
+        return $menu;
+    }
+
+    /**
+     * @throws ValidatorException
+     */
+    public function validateImportData(array $data)
+    {
+        $missingFields = [];
+
+        foreach (self::REQUIRED_FIELDS as $field) {
+            if (empty($data[$field])) {
+                $missingFields[] = $field;
+            }
+        }
+
+        if ($missingFields) {
+            throw new ValidatorException(
+                __('The following menu required import fields are missing: "%1".', implode('", "', $missingFields))
+            );
+        }
+    }
+
+    /**
+     * @param string $identifier
+     * @return string
+     */
+    private function getNewMenuIdentifier($identifier)
+    {
+        $menus = $this->getMenuListByIdentifier($identifier);
+        $identifiers = [];
+
+        foreach ($menus as $menu) {
+            $identifiers[$menu->getIdentifier()] = $menu->getId();
+        }
+
+        while (isset($identifiers[$identifier])) {
```

Comment:
so this for mutiple import of the same stuff will create:
mymenu
mymenu-1
mymenu-1-1
mymenu-1-1-1
mymenu-1-1-1-1

maybe we should think about making it
mymenu
mymenu-1
mymenu-2
and so on ? 😄

Reply:
This is actually a similar behavior in Magento 2 products SKUs, if you try to keep duplicating already duplicated products.

I applied some changes to improve it by incrementing the last part of a menu identifier.  
I made the identifier provided in CSV import file to always be the base for the new identifier, even if it already contains hyphens with numeric values.

If we try to dissect the import identifiers that already have hyphens further, which would be a bit more complicated,  
then we have to make sure that all identifiers with hyphens are fully/properly loaded from DB to avoid creating duplicate identifiers and avoid any unnecessary additional DB checks when generating a new identifier.

Note:  
In Magento 2 CMS module, pages and blocks duplicates' identifiers get generated using [`uniqid()`](https://github.com/magento/magento2/blob/2.4.1/app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php#L129).  
But the requirements for this menu import feature was to use Magento 2 products duplication approach,  
and append a numeric value at the end of the identifier.